app/controllers/vm_common.rb

```diff
@@ -1051,6 +1055,26 @@ def scan_histories
     end
   end
 
+  def parent_folder_id(vm)
```

Comment:
@jrafanie It determines parent id in tree. Unfortunately this solves it for three trees that are different (for example tree with Cloud VMs has Availability Zones(parent) displayed but tree with Cloud Templates doesn't).  `VmOrTemplate#parent_folder.id.to_cid` is not enough for me :)